### ISSUE TITLE
The generated serialization code for common enum is wrong

### ISSUE THREAD

**Issue by ArthurMa1978** [Azure/autorest.csharp#2709]
If an enum is common enum and the value is not string, then the generated serialization code is incorrect, like this

```
            if (Optional.IsDefined(AuthMethod))
            {
                writer.WritePropertyName("authMethod");
                writer.WriteStringValue(AuthMethod.Value.ToSerialString());
            }
```
https://github.com/Azure/azure-sdk-for-net/blob/ae614e3f5288b603a5e0060b7c9fe06386664fa1/sdk/botservice/Azure.ResourceManager.BotService/src/Generated/Models/EmailChannelProperties.Serialization.cs#L23

Here the `ToSerialString()` not return a string value, then the code can't be compiled. 
We should fix the `WriteEnumSerialization` function to generate right code.

**Comment by archerzz** [comment]
`AuthMethod` here is of type [EmailChannelAuthMethod](https://github.com/Azure/azure-sdk-for-net/blob/ae614e3f5288b603a5e0060b7c9fe06386664fa1/sdk/botservice/Azure.ResourceManager.BotService/src/Generated/Models/EmailChannelAuthMethod.cs#LL11C22-L11C22) which is a fixed int enum by [its swagger definition](https://github.com/Azure/azure-rest-api-specs/blob/af1be2677e619e483210064ff658e62ec25053aa/specification/botservice/resource-manager/Microsoft.BotService/preview/2022-06-15-preview/botservice.json#L2173).

This issue should be fixed in #2591

**Comment by archerzz** [comment]
Support of fixed int enum has been resolved in https://github.com/Azure/autorest.csharp/pull/2818

I did a local test, and the `EmailChannelAuthMethod` was correctly generated.
